lib/components/Upload.js

```diff
@@ -338,23 +377,20 @@ export default class Upload extends Component {
 
     if (upload.successful) {
       let dataDownloadLink = null;
-      if (__DEBUG__ && (!_.isEmpty(this.props.upload.data) && Array.isArray(this.props.upload.data))) {
-        let filename = 'uploader-processed-records.json';
-        let jsonData = JSON.stringify(this.props.upload.data, undefined, 4);
-        let blob = new Blob([jsonData], {type: 'text/json'});
-        let dataHref = URL.createObjectURL(blob);
-        dataDownloadLink = (
-          <a href={dataHref}
-            className={styles.dataDownloadLink}
-            download={filename}
-            data-downloadurl={['text/json', filename, dataHref].join(':')}>
-            Download POST data
-          </a>
-        );
+      if (__DEBUG__ && this.props.upload.data) {
```

Comment:
Perhaps throw the `!_.isEmpty` check back in? Again, not sure what's guaranteed here...been too long since I was elbow deep in this stuff.